imageio/imageio-metadata/src/main/java/com/twelvemonkeys/imageio/metadata/tiff/TIFFReader.java

```diff
@@ -135,7 +135,9 @@ protected Directory readDirectory(final ImageInputStream pInput, final long pOff
                 CompoundDirectory next = (CompoundDirectory) readDirectory(pInput, nextOffset, true);
 
                 for (int i = 0; i < next.directoryCount(); i++) {
-                    ifds.add((IFD) next.getDirectory(i));
+                    if(next.getDirectory(i).size() > 0) {
```

Comment:
Didn't think of that... :-/

Maybe the best behavior would be to leave empty sub-IFDs as is, but only skip empty linked IFDs (as they make no sense) similar to what you wrote initially. Just add a comment on why this (`size == 0`) may happen, so that we can clean it up later. There's a TODO that suggests we should rewrite the method to *not* handle linked IFD parsing at all in the method, might look at both at the same time.